File: dsa/1-introduction/firstProblem.js
// Problem
// Create an array with 6 students names, after that create a function which takes 2 parameters (allStudents and studentName) iterate over all students and find that specific user "studentName"


// Take 1
console.log('Take 1')

const studentsList = ['charizard', 'Orion', 'Nyx', 'ruki', 'anya', "charizard"]

function findStudentsByName(allStudents, studentName) {
    for (let i = 0; i < allStudents.length; i++) {
        const cleanStudentName = studentName.toLowerCase()

        if (allStudents[i].toLowerCase() === cleanStudentName) {
            console.log(`${cleanStudentName} found.`)
            return;
        }
    }

    console.log(`${studentName} is not on the list.`);
}

findStudentsByName(studentsList, 'CharizarD')



// Take 2
console.log('\nTake 2')

const studentsList2 = ['charizard', 'Orion', 'Nyx', 'ruki', 'anya', "charizard"]

function findStudentsByName2(allStudents, studentName) {
    const cleanStudentName = studentName.toLowerCase()
    return allStudents.find(student => student.toLowerCase() === cleanStudentName)
}

const foundStudent = findStudentsByName2(studentsList2, 'RUKI')

foundStudent !== undefined ? console.log(`${foundStudent} found.`) : console.log(`The student is not on the list.`)

File: dsa/1-introduction/first-problem.py
# Problem
# Create an array with 6 students names, after that create a function which takes 2 parameters (allStudents and studentName) iterate over all students and find that specific user "studentName"


# Take 1
print("Take 1")

students_list = {"charizard", "Orion", "Nyx", "ruki", "anya", "charizard"}


def find_students_by_name(allStudents, studentName):
    cleanStudentName = studentName.lower()

    for student in allStudents:
        if student.lower() == cleanStudentName:
            print(f"{cleanStudentName} found.")
            return

    print(f"{studentName} is not on the list.")


find_students_by_name(students_list, "CharizarD")


# Take 2
print("\nTake 2")


def find_students_by_name_2(allStudents, studentName):
    cleanStudentName = studentName.lower()
    allStudentsCleaned = set()

    for student in allStudents:
        allStudentsCleaned.add(student.lower())

    res = (
        f"{cleanStudentName} found."
        if cleanStudentName in allStudentsCleaned
        else f"{studentName} is not on the list."
    )

    print(res)


find_students_by_name_2(students_list, "RUKIS")
